scripts/yggdrasil/install

```diff
@@ -43,19 +43,33 @@ fi
 sudo sed -i "s/IfName: auto/IfName: ygg0/" /etc/yggdrasil.conf 
 
 # Change port to 12345 for firewall
-YGGDRASIL_PORT=$(sudo cat /etc/yggdrasil.conf | grep \ Listen: | awk  '{print $2}' | tr -d \")
-YGGDRASIL_PORT=$(printf "%q" "$YGGDRASIL_PORT") # Escape for sed
-sudo sed  -i "s/$YGGDRASIL_PORT/\[::\]:12345/" /etc/yggdrasil.conf
+sudo sed -i "s/Listen: \[\]/Listen\: [\"tcp:\/\/\[::\]:12345\"\]/" /etc/yggdrasil.conf
 
+sudo systemctl enable yggdrasil.service
+sudo systemctl start yggdrasil.service 
 
-# Set node name
-if [ -z "${NEWHOSTNAME}" ]; then 
-    NEWHOSTNAME=$(cat /etc/hostname)
+# Dont count yggdrasill ips as internet ip addresses
+if [ -f "/etc/gai.conf" ]; then
+  sudo sed -i '/label 200::/d' /etc/gai.conf
+  sudo sed -i '/label 300::/d' /etc/gai.conf
+  echo "label 200::/7 6" | sudo tee --append /etc/gai.conf
+  echo "label 300::/7 6" | sudo tee --append /etc/gai.conf
 fi
-sudo sed -i "\$i\ \ NodeInfo:\n  {\n   name: ${NEWHOSTNAME}\n  }\n"  /etc/yggdrasil.conf
 
-sudo systemctl enable yggdrasil.service
-sudo systemctl start yggdrasil.service 
 
 # Allow nginx access to yggdrasil socket when installed
 sudo usermod -a -G yggdrasil www-data || true
+
+# Define new hostname
+if [ -z "${NEWHOSTNAME}" ]; then
+   NEWHOSTNAME=$(sudo ifconfig ygg0 | grep -E 'inet6 2[0-9a-fA-F]{2}:' | awk '{print $2}' | sed 's/[",]//g' | sed "s/.*:/$MESH_NAME-Y/g")
```

Comment:
This definition will work, as opposed to the one I commented about [here](https://github.com/tomeshnet/prototype-cjdns-pi/pull/316/files#diff-54f569d767b5606b10c654a5ff1b63e3R128).

Reply:
hostname creation/generation should eventually be moved into a separate module.  

This is a very "tomesh node" specific thing and as i was thinking about making these modules more generic this is one example of somthing that a tomesh node needs but general people dont.

If you have CJDNS - generate a hostname with a CJDNS suffix
If you have  YGGDRASIL  and CJDNS did not set a host name, set a Yggdrasill suffix

This was done like this for now to remove CJDNS dependencies in building the prototype

Reply:
Yeah, I get that I guess. It's not great, but it can be fixed later ofc.

Reply:
https://github.com/tomeshnet/prototype-cjdns-pi/issues/372